src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -817,21 +842,50 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
       }
     }
 
+    this._saving = true;
     this._validationErrors = undefined;
+
     try {
       await saveAutomationConfig(this.hass, id, this._config!);
+
+      if (this._entityRegistryUpdate !== undefined) {
+        let entityId = this._entityId;
+
+        // wait for automation to appear in entity registry when creating a new automation
+        if (!entityId) {
+          await new Promise<void>((resolve) => {
+            setTimeout(resolve, 1000);
+          });
```

Comment:
Maybe something like this?

```suggestion
	  const automation = await new Promise((resolve) => {
	    this._entityRegCreated = resolve;
	  });
```

```ts
protected willUpdate(changedProps) {
  if (this._entityRegCreated && changedProps.has("entityRegistry")) {
    const automation = this.entityRegistry.find(
        (entity: EntityRegistryEntry) => entity.unique_id === id
      );
    if (automation) {
    	this._entityRegCreated(automation);
    	this._entityRegCreated = undefined;
    }
  }
}
```

Reply:
Yes, this what I had, worked well but the problem is, that the second `entity_registry_updated` of the `updateEntityRegistryEntry` will be missed, as `subscribeEntityRegistryUpdates` is debounced (https://github.com/home-assistant/frontend/blob/e9b2a8341142543941b0f5c366a8ab354c7d79d4/src/data/entity_registry.ts#L252).

In most cases the `updateEntityRegistryEntry` will be call just a few milliseconds after the automation save.

Reply:
Okay, then I'm going to add the suggestion 👍